src/test/kotlin/org/rust/lang/core/resolve/RsMarcoResolveTest.kt

```diff
@@ -0,0 +1,56 @@
+package org.rust.lang.core.resolve
+
+class RsMacroResolveTest : RsResolveTestBase() {
+    fun `test resolve simple matching with multiple pattern definition`() = checkByCode("""
+        macro_rules! test {
+            (${'$'}test:expr) => (
```

Comment:
Gosh, this `${'$'}` thing looks ugly :(

I wonder if this would look a tiny bit better if we define

```
private val `$` = '$'
```
at the file level and use

```
$`$`
```

instead?